vllm/reasoning/abs_reasoning_parsers.py

```diff
@@ -125,14 +119,16 @@ def get_reasoning_parser(cls, name) -> type:
         if name in cls.reasoning_parsers:
             return cls.reasoning_parsers[name]
 
-        raise KeyError(f"reasoning helper: '{name}' not found in "
-                       "reasoning_parsers")
+        raise KeyError(
+            f"reasoning helper: '{name}' not found in reasoning_parsers")
 
     @classmethod
-    def _register_module(cls,
-                         module: type,
-                         module_name: Optional[Union[str, list[str]]] = None,
-                         force: bool = True) -> None:
+    def _register_module(
+        cls,
+        module: type,
+        module_name: Optional[Union[str, list[str]]] = None,
+        force: bool = True,
```

Comment:
nit: This seems to be the `strict` behaviour. I would prefer renaming this variable to `strict` here.

also iiuc this is internal anyways, should we just make it strict by default?

I dont' really see the reason for user overriding this any time soon (given that we are maintaining this anw)?

Reply:
nit: for register_module I think would be nice if we can transform the target class into a dataclass, so that we can enforce start_token and end_token string

And setup `start_token_id` and `end_token_id` accordingly based on the vocab

Reply:
ditto. This is copied from the tool parser, maybe we could have another PR to change both of them?